src/System.Drawing.Common/src/System/Drawing/Imaging/ImageFormat.cs

```diff
@@ -161,16 +161,16 @@ internal ImageCodecInfo FindEncoder()
         /// </summary>
         public override string ToString()
         {
-            if (this == s_memoryBMP) return "MemoryBMP";
-            if (this == s_bmp) return "Bmp";
-            if (this == s_emf) return "Emf";
-            if (this == s_wmf) return "Wmf";
-            if (this == s_gif) return "Gif";
-            if (this == s_jpeg) return "Jpeg";
-            if (this == s_png) return "Png";
-            if (this == s_tiff) return "Tiff";
-            if (this == s_exif) return "Exif";
-            if (this == s_icon) return "Icon";
+            if (this.Guid == s_memoryBMP.Guid) return "MemoryBMP";
+            if (this.Guid == s_bmp.Guid) return "Bmp";
+            if (this.Guid == s_emf.Guid) return "Emf";
+            if (this.Guid == s_wmf.Guid) return "Wmf";
+            if (this.Guid == s_gif.Guid) return "Gif";
+            if (this.Guid == s_jpeg.Guid) return "Jpeg";
+            if (this.Guid == s_png.Guid) return "Png";
+            if (this.Guid == s_tiff.Guid) return "Tiff";
+            if (this.Guid == s_exif.Guid) return "Exif";
+            if (this.Guid == s_icon.Guid) return "Icon";
```

Comment:
Can you add tests for this?

Reply:
Sure. We already have one: https://github.com/dotnet/corefx/blob/master/src/System.Drawing.Common/tests/Imaging/ImageFormatTests.cs#L80 but I'm planning on adding one that uses Image.FromFile since that is when ToString was returning unexpected values.

Reply:
Thanks. I called it out because when there's a bug fix like this, I generally expect to either see a disabled test getting enabled or a new test getting added, something that would fail prior to the change and succeed after.

Reply:
I added the test.